graspologic/cluster/divisive_cluster.py

```diff
@@ -0,0 +1,272 @@
+# Copyright (c) Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+import numpy as np
+from sklearn.base import BaseEstimator
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+
+from anytree import NodeMixin
+
+from .autogmm import AutoGMMCluster
+from .kclust import KMeansCluster
+
+
+def _check_common_inputs(min_components, max_components, cluster_kws):
+    if not isinstance(min_components, int):
+        raise TypeError("min_components must be an int")
+    elif min_components < 1:
+        raise ValueError("min_components must be > 0")
+
+    if not isinstance(max_components, int):
+        raise TypeError("max_components must be an int")
+    elif max_components < 1:
+        raise ValueError("max_components must be > 0")
+    elif max_components < min_components:
+        raise ValueError("max_components must be >= min_components")
+
+    if not isinstance(cluster_kws, dict):
+        raise TypeError("cluster_kws must be a dict")
+
+
+class DivisiveCluster(NodeMixin, BaseEstimator):
+    """
+    Recursively clusters data based on a chosen clustering algorithm.
+    This algorithm implements a "divisive" or "top-down" approach.
+
+    Parameters
+    ----------
+    cluster_method : str {"gmm", "kmeans"}, defaults to "gmm".
+        The clustering method chosen to apply.
+    min_components : int, defaults to 1.
+        The minimum number of mixture components/clusters to consider
+        for the first split if "gmm" is selected as cluster_method;
+        and is set to 1 for later splits.
+        If "kmeans" is selected, it is set to 2 for all splits.
+    max_components : int, defaults to 2.
+        The maximum number of mixture components/clusters to consider
+        at each split.
+    min_split : int, defaults to 1.
+        The minimum size of a cluster for it to be considered to split again.
+    max_level : int, defaults to 4.
+        The maximum number of times to recursively cluster the data.
+    delta_criter : float, non-negative, defaults to 0.
```

Comment:
@tliu68 can you confirm/deny the non-negative part being necessary here, and make any appropriate changes